tests/affiliates/affiliates_integration.test.js

```diff
@@ -81,6 +82,8 @@ contract("Affiliates", (accounts) => {
     let referrerFee;
 
     async function deploymentAndInitFixture(_wallets, _provider) {
+        // need to deploy the mutex in the initialization. Otherwise, the global reentrancy prevetion will not be working & throw an error
```

Comment:
```suggestion  
        // Need to deploy the mutex in the initialization. Otherwise, the global reentrancy prevention will not be working & throw an error.
```